### DISCUSSION TITLE
[react-markdown] Pass custom props to custom components using plugin

### DISCUSSION THREAD

**Discussion by siliconeidolon** [remarkjs/remark#771] [Q&A] [answered]
I'm wondering if it's possible to pass custom props to my components after processing the tree with my own plugin. 

eg: I have my own Paragraph component and I'd like to pass an alignment prop based on what's discovered in the markdown. It seems that the `properties` object passed to the components has been stripped of any key-values I add to it in a plugin by the time it reaches the component. However, I am able to mutate the text value of the paragraph.

Here's a Codesandbox with a simple reproduction of what I'm trying to do: https://codesandbox.io/s/react-markdown-custom-props-wmlsf?file=/src/App.js 

```
const paragraphPlugin = () => {
  return (tree) => {
    visit(tree, "paragraph", (node) => {
      node.properties = { "data-custom-prop": "test" }; // This property is added to the node, but it never reaches the component
      visit(node, "text", (textNode) => {
        textNode.value = textNode.value + " mutated by a plugin.";
      });
    });
  };
};
```

Any advice is appreciated.

**Comment by wooorm** [comment, marked as answer]
Don’t set `properties` on mdast nodes. I’m not sure where you got that idea from, but it doesn’t work: markdown nodes don’t have a “properties”.

As you want to work with HTML, use a rehype plugin instead of a remark plugin. rehype plugins deal with an HTML ast. With elements. That do have properties.

**Reply by siliconeidolon** [reply]
Thanks, I'll look into creating a Rehype plugin instead. 

Update: the [codesandbox](https://codesandbox.io/s/react-markdown-custom-props-wmlsf?file=/src/App.js) is updated with a working example, if anyone is curious.
